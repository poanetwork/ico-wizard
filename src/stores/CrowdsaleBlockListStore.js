import { observable, action } from 'mobx';

class CrowdsaleBlockListStore {

	@observable crowdsaleBlockList;
	
	constructor(crowdsaleBlockList = []) {
    this.crowdsaleBlockList = crowdsaleBlockList;
  }

  @action addCrowdsaleBlock = (crowdsaleBlock) => {
    this.crowdsaleBlockList.push(crowdsaleBlock)
	}
	
	@action setCrowdsaleBlockProperty = (index, property, value) => {
		let newCrowdsaleBlock = {...this.crowdsaleBlockList[index]}
		newCrowdsaleBlock[property] = value
    this.crowdsaleBlockList[index] = newCrowdsaleBlock;
	}
	
	@action removeCrowdsaleBlock = (index) => {
		this.crowdsaleBlockList.splice(index,1)
	}

}

const crowdsaleBlockListStore = new CrowdsaleBlockListStore();

export default crowdsaleBlockListStore;
export { CrowdsaleBlockListStore };
