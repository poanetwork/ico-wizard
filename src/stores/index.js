import contractStore from './ContractStore';
import pricingStrategyStore from './PricingStrategyStore';
import reservedTokenStore from './ReservedTokenStore';
import reservedTokenInputStore from './ReservedTokenInputStore';
import reservedTokenElementStore from './ReservedTokenElementStore';
import stepThreeValidationStore from './StepThreeValidationStore';
import stepTwoValidationStore from './StepTwoValidationStore';
import tierStore from './TierStore';
import tokenStore from './TokenStore';
import web3Store from './Web3Store';
import tierCrowdsaleListStore from './TierCrowdsaleListStore'
import crowdsaleBlockListStore from './CrowdsaleBlockListStore'

export {
    tierCrowdsaleListStore,
    crowdsaleBlockListStore,
    contractStore,
    pricingStrategyStore,
    reservedTokenStore,
    reservedTokenInputStore,
    reservedTokenElementStore,
    stepThreeValidationStore,
    stepTwoValidationStore,
    tierStore,
    tokenStore,
    web3Store
};
