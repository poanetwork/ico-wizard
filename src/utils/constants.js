export const defaultState = { 
	contracts: { 
		token: {}, 
		crowdsale: {} 
	}, 
	token: {},
	crowdsale: {}
}
