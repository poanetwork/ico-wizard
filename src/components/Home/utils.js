import { DOWNLOAD_STATUS } from '../../utils/constants'
import { getNetworkVersion } from '../../utils/blockchainHelpers'
import { getCrowdsaleAssets } from '../../stores/utils'
import logdown from 'logdown'

const logger = logdown('TW:home:utils')

export const reloadStorage = async props => {
  let { generalStore, contractStore } = props

  try {
    if (!contractStore || !generalStore) {
      throw new Error('There is no stores to set')
    }
    contractStore.setProperty('downloadStatus', DOWNLOAD_STATUS.PENDING)

    // General store, check network
    let networkID = await getNetworkVersion()
    generalStore.setProperty('networkID', networkID)

    // Contract store, get contract and abi
    await getCrowdsaleAssets(networkID)
    contractStore.setProperty('downloadStatus', DOWNLOAD_STATUS.SUCCESS)
    return true
  } catch (e) {
    logger.error('Error downloading contracts', e)
    if (contractStore) {
      contractStore.setProperty('downloadStatus', DOWNLOAD_STATUS.FAILURE)
    }
    throw e
  }
}
