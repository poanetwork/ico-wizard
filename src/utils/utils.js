import { TOAST } from './constants'
import queryString from 'query-string'
import { BigNumber } from 'bignumber.js'
import logdown from 'logdown'
import Web3 from 'web3'
import moment from 'moment'
import { fetchFile } from './fetchFile'
import { isObservableArray } from 'mobx'

const logger = logdown('TW:utils:utils')

export const getQueryVariable = variable => {
  return queryString.parse(window.location.search)[variable]
}

export const isExecIDValid = execID => /^0x[a-f0-9]{64}$/i.test(execID)

export const isAddressValid = address => Web3.utils.isAddress(address)

export const getExecIDFromQuery = () => {
  const execID = getQueryVariableExecId()
  logger.log('getExecID:', execID)
  return isExecIDValid(execID) ? execID : null
}

export const getAddrFromQuery = () => {
  const addr = getQueryVariableAddr()
  logger.log('getAddr:', addr)
  return isAddressValid(addr) ? addr : null
}

export const getQueryVariableAddr = () => {
  return getQueryVariable('addr')
}

export const getQueryVariableExecId = () => {
  return getQueryVariable('exec-id')
}

export const isNetworkIDValid = networkID => /^[0-9]+$/.test(networkID)

export const getNetworkID = () => {
  const networkID = getQueryVariable('networkID')
  return isNetworkIDValid(networkID) ? networkID : null
}

export const setFlatFileContentToState = async file => {
  return await fetchFile(file)
}

export const dateToTimestamp = date => new Date(date).getTime()

export const validateTier = tier => typeof tier === 'string' && tier.length > 0 && tier.length < 30

export const validateSupply = supply => isNaN(Number(supply)) === false && Number(supply) > 0

export const validateTime = time => dateToTimestamp(time) > Date.now()

export const validateLaterTime = (laterTime, previousTime) => dateToTimestamp(laterTime) > dateToTimestamp(previousTime)

export const validateLaterOrEqualTime = (laterTime, previousTime) =>
  dateToTimestamp(laterTime) >= dateToTimestamp(previousTime)

export const toFixed = x => {
  if (Math.abs(x) < 1.0) {
    let e = parseInt(x.toString().split('e-')[1], 10)
    if (e) {
      x *= Math.pow(10, e - 1)
      x = '0.' + new Array(e).join('0') + x.toString().substring(2)
    }
  } else {
    let e = parseInt(x.toString().split('+')[1], 10)
    if (e > 20) {
      e -= 20
      x /= Math.pow(10, e)
      x += new Array(e + 1).join('0')
    }
  }
  return x
}

export const toast = {
  msg: {},
  showToaster: function({ type = TOAST.TYPE.INFO, message = '', options = {} }) {
    if (!message) {
      return
    }

    if (typeof this.msg[type] === 'function') {
      this.msg[type](message, options)
    } else {
      return
    }
  }
}

export const gweiToWei = x => parseInt(x * 1000000000, 10)

export const weiToGwei = x => x / 1000000000

export const countDecimalPlaces = num => {
  /*
    (?:
      \.
      (\d+)  First captured group: decimals after the point but before the e
    )?
    (?:
      [eE]
      ([+-]?\d+)  Second captured group: exponent used to adjust the count
    )?
    $
  */
  const match = ('' + num).match(/(?:\.(\d+))?(?:[eE]([+-]?\d+))?$/)

  if (!match[0] && !match[1] && !match[2]) return 0

  const digitsAfterDecimal = match[1] ? match[1].length : 0
  const adjust = match[2] ? +match[2] : 0

  return Math.max(0, digitsAfterDecimal - adjust)
}

export const acceptPositiveIntegerOnly = value => {
  if (typeof value === 'number') value = String(value)
  if (typeof value !== 'string') return ''

  return String(value).match(/^(\d*)/)[1]
}

export const removeTrailingNUL = ascii => ascii.replace(/\x00+$/, '')

export const truncateStringInTheMiddle = (str, strLength = 50, strPositionStart = 24, strPositionEnd = 25) => {
  if (typeof str === 'string' && str.length > strLength) {
    return `${str.substr(0, strPositionStart)}...${str.substr(str.length - strPositionEnd, str.length)}`
  }
  return str
}

/**
 * Converts the value passed to a BigNumber instance
 * @param {*} value - A number representation
 * @param {boolean} [force=true] - If set to false will return 'undefined' when value is not a number or a string
 * representation of a number.
 * @returns {BigNumber|undefined}
 */
export const toBigNumber = (value, force = true) => {
  BigNumber.set({ DECIMAL_PLACES: 18 })

  if (isNaN(value) || value === '' || value === null) {
    if (force) {
      return new BigNumber(0)
    } else {
      return undefined
    }
  } else {
    return new BigNumber(value)
  }
}

/**
 * Sleep function like C
 * @param ms
 * @returns {Promise}
 */
export const sleep = async ms => {
  return new Promise(resolve => setTimeout(resolve, ms))
}

export const clearStorage = props => {
  // Generate of stores to clear
  const toArray = ({
    generalStore,
    contractStore,
    crowdsaleStore,
    gasPriceStore,
    deploymentStore,
    reservedTokenStore,
    stepTwoValidationStore,
    tierStore,
    tokenStore
  }) => {
    return [
      generalStore,
      contractStore,
      crowdsaleStore,
      gasPriceStore,
      deploymentStore,
      reservedTokenStore,
      stepTwoValidationStore,
      tierStore,
      tokenStore
    ]
  }

  const storesToClear = toArray(props)
  for (let storeToClear of storesToClear) {
    if (storeToClear && typeof storeToClear.reset === 'function') {
      logger.log('Store to be cleared:', storeToClear.constructor.name)
      storeToClear.reset()
    }
  }
}

export const navigateTo = data => {
  const { history, location, params = '', fromLocation } = data
  const path = convertLocationToPath(location)

  if (path === null) {
    throw new Error(`invalid location specified: ${location}`)
  }

  if (!history || !(typeof history === 'object')) {
    throw new Error(`invalid history object: ${history}`)
  }

  history.push({
    pathname: `${path}${params}`,
    state: {
      fromLocation: fromLocation
    }
  })

  return true
}

export const convertLocationToPath = location => {
  return (
    {
      home: '/',
      stepOne: '1',
      stepTwo: '2',
      stepThree: '3',
      stepFour: '4',
      manage: 'manage',
      crowdsales: 'crowdsales',
      contribute: 'contribute'
    }[location] || null
  )
}

export const goBack = data => {
  const { history } = data

  if (!history || !(typeof history === 'object')) {
    throw new Error(`invalid history object: ${history}`)
  }

  if (typeof history.goBack === undefined) {
    throw new Error(`invalid goBack function`)
  }

  history.goBack()

  return true
}

export const goBackMustBeEnabled = data => {
  const { history } = data

  if (!history || !(typeof history === 'object')) {
    throw new Error(`invalid history object: ${history}`)
  }
  const length = history.length || 0
  return length > 1
}

export const convertDateToLocalTimezoneInUnix = dateToConvert => {
  const dateConvertUnix = moment(dateToConvert)
    .local()
    .unix()
  return dateConvertUnix * 1000
}

export const convertDateToUTCTimezoneToDisplay = dateToConvert => {
  return moment(dateToConvert)
    .utc()
    .format('YYYY-MM-DD HH:mm (z ZZ)')
}

export const getContractBySourceType = (sourceType, isMintedCappedCrowdsale, { DutchProxy, MintedCappedProxy }) => {
  const parseContent = content => (isObservableArray(content) ? JSON.stringify(content.slice()) : content)
  return isMintedCappedCrowdsale ? parseContent(MintedCappedProxy[sourceType]) : parseContent(DutchProxy[sourceType])
}

export const getSourceTypeTitle = sourceType => {
  const sourceTypeName = {
    abi: 'ABI',
    bin: 'Creation Code',
    src: 'Source Code'
  }

  return `Crowdsale Proxy Contract ${sourceTypeName[sourceType]}`
}

export const updateProxyContractInfo = ({ contractAddress }, { web3Store, contractStore, crowdsaleStore }, params) => {
  const { web3 } = web3Store
  const encoded = web3.eth.abi.encodeParameters(['address', 'bytes32', 'address', 'bytes32'], params)

  contractStore.setContractProperty(crowdsaleStore.proxyName, 'addr', contractAddress.toLowerCase())
  contractStore.setContractProperty(crowdsaleStore.proxyName, 'abiEncoded', encoded.slice(2))
}

export const downloadFile = (data, filename, mimetype) => {
  if (!data) {
    return
  }

  const blob = data.constructor !== Blob ? new Blob([data], { type: mimetype || 'application/octet-stream' }) : data

  if (navigator.msSaveBlob) {
    navigator.msSaveBlob(blob, filename)
    return
  }

  const lnk = document.createElement('a')
  const url = window.URL
  let objectURL

  if (mimetype) {
    lnk.type = mimetype
  }

  lnk.download = filename || 'untitled'
  lnk.href = objectURL = url.createObjectURL(blob)
  lnk.dispatchEvent(new MouseEvent('click'))
  setTimeout(url.revokeObjectURL.bind(url, objectURL))
}

export const uniqueElementsBy = (arr, fn) =>
  arr.reduce((acc, v) => {
    if (!acc.some(x => fn(v, x))) acc.push(v)
    return acc
  }, [])

export const pad = (num, size) => {
  let s = num + ''
  while (s.length < size) s = '0' + s
  return s
}
