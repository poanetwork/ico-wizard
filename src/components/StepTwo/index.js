import React, { Component } from 'react'
import logdown from 'logdown'
import setFieldTouched from 'final-form-set-field-touched'
import { Form } from 'react-final-form'
import { NAVIGATION_STEPS } from '../../utils/constants'
import { SectionInfo } from '../Common/SectionInfo'
import { StepNavigation } from '../Common/StepNavigation'
import { StepTwoForm } from './StepTwoForm'
import { checkWeb3 } from '../../utils/blockchainHelpers'
import { inject, observer } from 'mobx-react'
import { navigateTo, goBack, goBackMustBeEnabled } from '../../utils/utils'

const { TOKEN_SETUP } = NAVIGATION_STEPS
const logger = logdown('TW:stepTwo:index')

@inject('tokenStore', 'crowdsaleStore', 'reservedTokenStore')
@observer
export class StepTwo extends Component {
  state = {
    tokenValues: {},
    reload: false,
    backButtonTriggered: false, //Testing purposes
    nextButtonTriggered: false, //Testing purposes
    goBackEnabledTriggered: false //Testing purposes
  }

  async componentDidMount() {
    await checkWeb3()

    logger.log('Component did mount')
    window.addEventListener('beforeunload', this.onUnload)

    const tokenValues = this.load()
    logger.log('Token Values', tokenValues)
    this.setState({ tokenValues: tokenValues })
  }

  load() {
    const { tokenStore, crowdsaleStore, reservedTokenStore } = this.props
    const isEmpty = crowdsaleStore.isDutchAuction ? tokenStore.checkIsEmptyDutch : tokenStore.checkIsEmptyMinted

    if (isEmpty) {
      tokenStore.addTokenSetup()
    } else {
      this.setState({
        reload: true
      })
    }

    const token = crowdsaleStore.isDutchAuction ? tokenStore.tokenDutchStructure : tokenStore.tokenMintedStructure
    reservedTokenStore.applyDecimalsToTokens(token.decimals)
    return token
  }

  goNextStep = () => {
    try {
      this.setState({
        nextButtonTriggered: true
      })
      navigateTo({
        history: this.props.history,
        location: 'stepThree',
        fromLocation: 'stepTwo'
      })
    } catch (err) {
      logger.log('Error to navigate', err)
    }
  }

  goBack = () => {
    try {
      this.setState({
        backButtonTriggered: true
      })
      goBack({
        history: this.props.history,
        location: '/stepOne'
      })
    } catch (err) {
      logger.log('Error to navigate', err)
    }
  }

  goBackEnabled = () => {
    let goBackEnabled = false
    try {
      this.setState({
        goBackEnabledTriggered: true
      })
      goBackEnabled = goBackMustBeEnabled({ history: this.props.history })
      logger.log(`Go back is enabled ${goBackEnabled}`)
    } catch (err) {
      logger.log(`There is an error trying to set enable/disable on back button`)
    }
    return goBackEnabled
  }

  onUnload = e => {
    logger.log('On unload')
    e.returnValue = 'Are you sure you want to leave?'
  }

  componentWillUnmount() {
    logger.log('Component unmount')
    window.removeEventListener('beforeunload', this.onUnload)
  }

  render() {
    return (
      <div>
        <section className="lo-MenuBarAndContent" ref="two">
          <StepNavigation activeStepTitle={TOKEN_SETUP} />
          <div className="st-StepContent">
            <SectionInfo
              description="Configure properties for your token. Created token contract will be ERC-20 compatible."
              stepNumber="2"
              title="Token Setup"
            />
            <Form
              component={StepTwoForm}
              id="tokenData"
              initialValues={this.state.tokenValues}
              mutators={{ setFieldTouched }}
              onSubmit={this.goNextStep}
              reload={this.state.reload}
              goBack={this.goBack}
              goBackEnabled={this.goBackEnabled}
            />
          </div>
        </section>
      </div>
    )
  }
}
