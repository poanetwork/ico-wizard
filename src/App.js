import React, {Component} from 'react';
import './assets/stylesheets/application.css';
import { Header, Footer, Home, stepOne, stepTwo, stepThree, stepFour, stepFive, Invest } from './components/index'
import { getQueryVariable } from './utils/utils'
import {
  BrowserRouter as Router,
  Route
} from 'react-router-dom'

class App extends Component {
  render() {
    var crowdsaleAddr = getQueryVariable("addr");
    return (
      <Router>
        <div>
          <Header/>
            <Route exact path="/" component={crowdsaleAddr?stepFive:Home}/>
            <Route path="/1" component={stepOne}/>
            <Route path="/2" component={stepTwo}/>
            <Route path="/3" component={stepThree}/>
            <Route path="/4" component={stepFour}/>
            <Route path="/crowdsale" component={stepFive}/>
            <Route path="/invest" component={Invest}/>
          <Footer/>
        </div>
      </Router>
    )
  }
}

export default App;
