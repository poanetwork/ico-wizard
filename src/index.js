import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import registerServiceWorker from './registerServiceWorker';
import { useStrict } from 'mobx';
import { Provider } from 'mobx-react';
import * as stores from './stores';
useStrict(true);


ReactDOM.render(
  <Provider { ...stores }>
    <App />
  </Provider>, 
document.getElementById('root'));
registerServiceWorker();
