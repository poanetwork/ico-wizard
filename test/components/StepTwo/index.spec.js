import React from 'react'
import Adapter from 'enzyme-adapter-react-15'
import { Provider } from 'mobx-react'
import { configure, mount, shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router'
import { StepTwo } from '../../../src/components/StepTwo/index'
import { tokenStore, crowdsaleStore, reservedTokenStore } from '../../../src/stores'
import { CROWDSALE_STRATEGIES } from '../../../src/utils/constants'

configure({ adapter: new Adapter() })

describe('StepTwo', () => {
  const stores = { tokenStore, crowdsaleStore, reservedTokenStore }
  const history = {
    push: jest.fn()
  }

  it(`should render StepTwo for Minted Capped strategy`, () => {
    // Given
    const component = renderer.create(
      <Provider {...stores}>
        <MemoryRouter initialEntries={['/', '1', '2']} initialIndex={2}>
          <StepTwo />
        </MemoryRouter>
      </Provider>
    )

    // When
    const tree = component.toJSON()

    // Then
    expect(tree).toMatchSnapshot()
  })

  it(`should render StepTwo for Dutch Auction strategy`, () => {
    // Given
    crowdsaleStore.setProperty('strategy', CROWDSALE_STRATEGIES.DUTCH_AUCTION)
    const component = renderer.create(
      <Provider {...stores}>
        <MemoryRouter initialEntries={['/', '1', '2']} initialIndex={2}>
          <StepTwo />
        </MemoryRouter>
      </Provider>
    )

    // When
    const tree = component.toJSON()

    // Then
    expect(tree).toMatchSnapshot()
  })

  it(`should click go back button`, () => {
    const wrapper = mount(
      <Provider {...stores}>
        <MemoryRouter initialEntries={['/', '1', '2']} initialIndex={2}>
          <StepTwo />
        </MemoryRouter>
      </Provider>
    )
    const stepTwoComponent = wrapper.find('StepTwo')

    // When
    stepTwoComponent.find('button.sw-ButtonBack').simulate('click')

    // Then
    expect(stepTwoComponent.instance().state.backButtonTriggered).toBeTruthy()
  })

  it(`should click next button`, () => {
    const wrapper = mount(
      <Provider {...stores}>
        <MemoryRouter initialEntries={['/', '1', '2']} initialIndex={2}>
          <StepTwo />
        </MemoryRouter>
      </Provider>
    )
    const stepTwoComponent = wrapper.find('StepTwo')

    // When
    stepTwoComponent.find('button.sw-ButtonContinue').simulate('click')

    setTimeout(() => {
      // Then
      expect(stepTwoComponent.instance().state.nextButtonTriggered).toBeTruthy()
    }, 2000)
  })

  it(`should execute goBack`, () => {
    const wrapper = mount(
      <Provider {...stores}>
        <MemoryRouter initialEntries={['/', '1', '2']} initialIndex={2}>
          <StepTwo />
        </MemoryRouter>
      </Provider>
    )
    // Then
    const stepTwoComponent = wrapper.find('StepTwo')
    const stepTwoComponentInstance = stepTwoComponent.instance()
    stepTwoComponentInstance.goBack()

    // When
    expect(stepTwoComponentInstance.state.backButtonTriggered).toBeTruthy()
  })

  it(`should execute goNextStep`, () => {
    const wrapper = mount(
      <Provider {...stores}>
        <MemoryRouter initialEntries={['/', '1', '2']} initialIndex={2}>
          <StepTwo />
        </MemoryRouter>
      </Provider>
    )
    // Then
    const stepTwoComponent = wrapper.find('StepTwo')
    const stepTwoComponentInstance = stepTwoComponent.instance()
    stepTwoComponentInstance.goNextStep()

    // When
    expect(stepTwoComponentInstance.state.nextButtonTriggered).toBeTruthy()
  })

  it(`should execute goBackEnabled`, () => {
    const wrapper = mount(
      <Provider {...stores}>
        <MemoryRouter initialEntries={['/', '1', '2']} initialIndex={2}>
          <StepTwo history={history} />
        </MemoryRouter>
      </Provider>
    )
    // Then
    const stepTwoComponent = wrapper.find('StepTwo')
    const stepTwoComponentInstance = stepTwoComponent.instance()
    stepTwoComponentInstance.goBackEnabled()

    // When
    expect(stepTwoComponentInstance.state.goBackEnabledTriggered).toBeTruthy()
  })
})
