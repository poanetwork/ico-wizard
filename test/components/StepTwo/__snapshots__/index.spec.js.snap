// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepTwo should render StepTwo for Dutch Auction strategy 1`] = `
<div>
  <section
    className="lo-MenuBarAndContent"
  >
    <div
      className="st-StepNavigation"
    >
      <a
        className="sw-LogoWhite"
        href="/"
      />
      <div
        className="st-StepNavigation_Container"
      >
        <div
          className="st-StepItem
        active  "
        >
          <div
            className="st-StepItem_Circle"
          />
          <div
            className="st-StepItem_Line"
          />
          <span
            className="st-StepItem_Text"
          >
            Crowdsale Strategy
          </span>
        </div>
        <div
          className="st-StepItem
        active  "
        >
          <div
            className="st-StepItem_Circle"
          />
          <div
            className="st-StepItem_Line"
          />
          <span
            className="st-StepItem_Text"
          >
            Token Setup
          </span>
        </div>
        <div
          className="st-StepItem
          "
        >
          <div
            className="st-StepItem_Circle"
          />
          <div
            className="st-StepItem_Line"
          />
          <span
            className="st-StepItem_Text"
          >
            Crowdsale Setup
          </span>
        </div>
        <div
          className="st-StepItem
          "
        >
          <div
            className="st-StepItem_Circle"
          />
          <div
            className="st-StepItem_Line"
          />
          <span
            className="st-StepItem_Text"
          >
            Publish
          </span>
        </div>
        <div
          className="st-StepItem
          "
        >
          <div
            className="st-StepItem_Circle"
          />
          <span
            className="st-StepItem_Text"
          >
            Crowdsale Page
          </span>
        </div>
      </div>
    </div>
    <div
      className="st-StepContent"
    >
      <div
        className="st-SectionInfo "
      >
        <div
          className="st-SectionInfo_Icon st-SectionInfo_Icon-step2"
        />
        <div
          className="st-SectionInfo_Text"
        >
          <h1
            className="st-SectionInfo_Title"
          >
            Token Setup
          </h1>
          <p
            className="st-SectionInfo_Description"
            dangerouslySetInnerHTML={
              Object {
                "__html": "Configure properties for your token. Created token contract will be ERC-20 compatible.",
              }
            }
          />
        </div>
      </div>
      <form
        className="st-StepContent_FormFullHeight"
        id="tokenData"
        onSubmit={[Function]}
      >
        <div
          className="sw-BorderedBlock sw-BorderedBlock-GeneralSettingsDutchAuction"
        >
          <div
            className="sw-InputField2 "
          >
            <div
              className="sw-FormControlTitle"
            >
              <label
                className="sw-FormControlTitle_Label"
              >
                Name
              </label>
              <div
                className="sw-FormControlTitle_Info"
                tabIndex="-1"
              >
                <span
                  className="sw-FormControlTitle_Tooltip"
                >
                  The name of your token. Will be used by Etherscan and other tokenbrowsers. Be afraid of trademarks.
                </span>
              </div>
            </div>
            <input
              autoComplete="off"
              className="sw-TextField"
              id="name"
              name="name"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Enter here"
              type="text"
              value=""
            />
          </div>
          <div
            className="sw-InputField2 "
          >
            <div
              className="sw-FormControlTitle"
            >
              <label
                className="sw-FormControlTitle_Label"
              >
                Ticker
              </label>
              <div
                className="sw-FormControlTitle_Info"
                tabIndex="-1"
              >
                <span
                  className="sw-FormControlTitle_Tooltip"
                >
                  The five letter ticker for your token. There are 11,881,376 combinations for 26 english letters. Be hurry.
                </span>
              </div>
            </div>
            <input
              autoComplete="off"
              className="sw-TextField"
              id="ticker"
              name="ticker"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Enter here"
              type="text"
              value=""
            />
          </div>
          <div
            className="sw-InputField2 "
          >
            <div
              className="sw-FormControlTitle"
            >
              <label
                className="sw-FormControlTitle_Label"
              >
                Decimals
              </label>
              <div
                className="sw-FormControlTitle_Info"
                tabIndex="-1"
              >
                <span
                  className="sw-FormControlTitle_Tooltip"
                >
                  Refers to how divisible a token can be, from 0 (not at all divisible) to 18 (pretty much continuous).
                </span>
              </div>
            </div>
            <input
              autoComplete="off"
              className="sw-TextField"
              disabled={false}
              id="decimals"
              max="18"
              min="0"
              name="decimals"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Enter here"
              type="number"
              value=""
            />
          </div>
          <div
            className="sw-InputField2 "
          >
            <div
              className="sw-FormControlTitle"
            >
              <label
                className="sw-FormControlTitle_Label"
              >
                Supply
              </label>
              <div
                className="sw-FormControlTitle_Info"
                tabIndex="-1"
              >
                <span
                  className="sw-FormControlTitle_Tooltip"
                >
                  The total supply of the token
                </span>
              </div>
            </div>
            <input
              autoComplete="off"
              className="sw-TextField"
              id="supply"
              min="0"
              name="supply"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Enter here"
              type="number"
              value=""
            />
          </div>
        </div>
        <div
          className="st-StepContent_Buttons"
        >
          <button
            className="sw-ButtonBack "
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Back
          </button>
          <button
            className="sw-ButtonContinue "
            disabled={true}
            type="submit"
          >
            <span
              className="sw-ButtonContinue_Text"
            >
              Continue
            </span>
          </button>
        </div>
      </form>
    </div>
  </section>
</div>
`;

exports[`StepTwo should render StepTwo for Minted Capped strategy 1`] = `
<div>
  <section
    className="lo-MenuBarAndContent"
  >
    <div
      className="st-StepNavigation"
    >
      <a
        className="sw-LogoWhite"
        href="/"
      />
      <div
        className="st-StepNavigation_Container"
      >
        <div
          className="st-StepItem
        active  "
        >
          <div
            className="st-StepItem_Circle"
          />
          <div
            className="st-StepItem_Line"
          />
          <span
            className="st-StepItem_Text"
          >
            Crowdsale Strategy
          </span>
        </div>
        <div
          className="st-StepItem
        active  "
        >
          <div
            className="st-StepItem_Circle"
          />
          <div
            className="st-StepItem_Line"
          />
          <span
            className="st-StepItem_Text"
          >
            Token Setup
          </span>
        </div>
        <div
          className="st-StepItem
          "
        >
          <div
            className="st-StepItem_Circle"
          />
          <div
            className="st-StepItem_Line"
          />
          <span
            className="st-StepItem_Text"
          >
            Crowdsale Setup
          </span>
        </div>
        <div
          className="st-StepItem
          "
        >
          <div
            className="st-StepItem_Circle"
          />
          <div
            className="st-StepItem_Line"
          />
          <span
            className="st-StepItem_Text"
          >
            Publish
          </span>
        </div>
        <div
          className="st-StepItem
          "
        >
          <div
            className="st-StepItem_Circle"
          />
          <span
            className="st-StepItem_Text"
          >
            Crowdsale Page
          </span>
        </div>
      </div>
    </div>
    <div
      className="st-StepContent"
    >
      <div
        className="st-SectionInfo "
      >
        <div
          className="st-SectionInfo_Icon st-SectionInfo_Icon-step2"
        />
        <div
          className="st-SectionInfo_Text"
        >
          <h1
            className="st-SectionInfo_Title"
          >
            Token Setup
          </h1>
          <p
            className="st-SectionInfo_Description"
            dangerouslySetInnerHTML={
              Object {
                "__html": "Configure properties for your token. Created token contract will be ERC-20 compatible.",
              }
            }
          />
        </div>
      </div>
      <form
        className="st-StepContent_FormFullHeight"
        id="tokenData"
        onSubmit={[Function]}
      >
        <div
          className="sw-BorderedBlock sw-BorderedBlock-GeneralSettingsWhitelistCapped"
        >
          <div
            className="sw-InputField2 "
          >
            <div
              className="sw-FormControlTitle"
            >
              <label
                className="sw-FormControlTitle_Label"
              >
                Name
              </label>
              <div
                className="sw-FormControlTitle_Info"
                tabIndex="-1"
              >
                <span
                  className="sw-FormControlTitle_Tooltip"
                >
                  The name of your token. Will be used by Etherscan and other tokenbrowsers. Be afraid of trademarks.
                </span>
              </div>
            </div>
            <input
              autoComplete="off"
              className="sw-TextField"
              id="name"
              name="name"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Enter here"
              type="text"
              value=""
            />
          </div>
          <div
            className="sw-InputField2 "
          >
            <div
              className="sw-FormControlTitle"
            >
              <label
                className="sw-FormControlTitle_Label"
              >
                Ticker
              </label>
              <div
                className="sw-FormControlTitle_Info"
                tabIndex="-1"
              >
                <span
                  className="sw-FormControlTitle_Tooltip"
                >
                  The five letter ticker for your token. There are 11,881,376 combinations for 26 english letters. Be hurry.
                </span>
              </div>
            </div>
            <input
              autoComplete="off"
              className="sw-TextField"
              id="ticker"
              name="ticker"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Enter here"
              type="text"
              value=""
            />
          </div>
          <div
            className="sw-InputField2 "
          >
            <div
              className="sw-FormControlTitle"
            >
              <label
                className="sw-FormControlTitle_Label"
              >
                Decimals
              </label>
              <div
                className="sw-FormControlTitle_Info"
                tabIndex="-1"
              >
                <span
                  className="sw-FormControlTitle_Tooltip"
                >
                  Refers to how divisible a token can be, from 0 (not at all divisible) to 18 (pretty much continuous).
                </span>
              </div>
            </div>
            <input
              autoComplete="off"
              className="sw-TextField"
              disabled={false}
              id="decimals"
              max="18"
              min="0"
              name="decimals"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Enter here"
              type="number"
              value=""
            />
          </div>
        </div>
        <div
          className="st-StepContent_Buttons"
        >
          <button
            className="sw-ButtonBack "
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Back
          </button>
          <button
            className="sw-ButtonContinue "
            disabled={true}
            type="submit"
          >
            <span
              className="sw-ButtonContinue_Text"
            >
              Continue
            </span>
          </button>
        </div>
      </form>
    </div>
  </section>
</div>
`;
